File: backend/app/models/portallicense.py
from ..db import db
from sqlalchemy import UUID


class Portallicense(db.Model):
    __tablename__ = 'portallicenses'
    guid = db.Column(UUID, primary_key=True, nullable=False, unique=True)
    id = db.Column(db.String(120), nullable=False, unique=True)
    name = db.Column(db.String(120), nullable=False)
    description = db.Column(db.String(255), nullable=True)
    level = db.Column(db.String(120), nullable=False)
    state = db.Column(db.String(120), nullable=False)
    maxusers = db.Column(db.Integer, nullable=False)

    def to_dict(self):
        return {
            "guid": self.guid,
            "id": self.id,
            "name": self.name,
            "description": self.description,
            "level": self.level,
            "state": self.state,
            "maxusers": self.maxusers
        }

    def __repr__(self):
        return f"<Portallicense {self.guid} - {self.name}>"

File: frontend/src/models/portallicense.ts
import {AbstractModel} from "./abstract-models.ts";

export interface PortalLicenseProps {
    guid: string;
    id: string;
    name: string;
    description?: string;
    level: string;
    state: string;
    maxusers: number;
    currentusers: number;
}

export class PortalLicense extends AbstractModel {
    class: string = 'PortalLicense';
    guid: string;
    id: string;
    name: string;
    description?: string;
    level: string;
    state: string;
    maxusers: number;
    currentusers: number;

    constructor(props: PortalLicenseProps) {
        super();
        this.guid = props.guid;
        this.id = props.id;
        this.name = props.name;
        this.description = props.description;
        this.level = props.level;
        this.state = props.state;
        this.maxusers = props.maxusers;
        this.currentusers = props.currentusers;
    }
}